src/tri.rs

```diff
@@ -30,38 +39,44 @@ where
     /// ```
     /// use ndarray::array;
     ///
-    /// let arr = array![[1, 2, 3], [4, 5, 6], [7, 8, 9]];
-    /// let res = arr.triu(0);
-    /// assert_eq!(res, array![[1, 2, 3], [0, 5, 6], [0, 0, 9]]);
+    /// let arr = array![
+    ///     [1, 2, 3],
+    ///     [4, 5, 6],
+    ///     [7, 8, 9]
+    /// ];
+    /// assert_eq!(
+    ///     arr.triu(0),
+    ///     array![
+    ///         [1, 2, 3],
+    ///         [0, 5, 6],
+    ///         [0, 0, 9]
+    ///     ]
+    /// );
     /// ```
     pub fn triu(&self, k: isize) -> Array<A, D>
     {
         if self.ndim() <= 1 {
             return self.to_owned();
         }
-        match is_layout_f(&self.dim, &self.strides) {
-            true => {
-                let n = self.ndim();
-                let mut x = self.view();
-                x.swap_axes(n - 2, n - 1);
-                let mut tril = x.tril(-k);
-                tril.swap_axes(n - 2, n - 1);
-
-                tril
-            }
-            false => {
-                let mut res = Array::zeros(self.raw_dim());
-                Zip::indexed(self.rows())
-                    .and(res.rows_mut())
-                    .for_each(|i, src, mut dst| {
-                        let row_num = i.into_dimension().last_elem();
-                        let lower = max(row_num as isize + k, 0);
-                        dst.slice_mut(s![lower..]).assign(&src.slice(s![lower..]));
-                    });
-
-                res
-            }
+        if is_layout_f(&self.dim, &self.strides) && !is_layout_c(&self.dim, &self.strides) {
+            let n = self.ndim();
+            let mut x = self.view();
+            x.swap_axes(n - 2, n - 1);
+            let mut tril = x.tril(-k);
```

Comment:
If all `k` values are accepted (maybe in the future), then k=`isize::MIN` has a problem here.

Reply:
Aiming for all `k` to be allowed, I've added a check for this optimization that skips it if `k == isize::MIN`.